File: redhawk/codegen/jinja/cpp/ports/templates/generic.provides.cpp
//% set classname = portgen.className()
${classname}::${classname}(std::string port_name, ${component.baseclass.name} *_parent) : 
Port_Provides_base_impl(port_name)
{
    parent = static_cast<${component.userclass.name} *> (_parent);
}

${classname}::~${classname}()
{
}
/*{% for operation in portgen.operations() %}*/

${operation.returns} ${classname}::${operation.name}(${operation.arglist})
{
    boost::mutex::scoped_lock lock(portAccess);
/*{% if operation.returns != 'void' %}*/
/*{%   if operation.returns == 'CORBA::Object_ptr' %}*/
    ${operation.returns} tmpVal = CORBA::Object::_nil();
/*{%   else %}*/
    ${operation.returns} tmpVal;
/*{%   endif %}*/
/*{% endif %}*/
    // TODO: Fill in this function
/*{% if operation.returns != 'void' %}*/

/*{%   if operation.returns == 'char*' %}*/
    return CORBA::string_dup(tmpVal);
/*{%   else %}*/
    return tmpVal;
/*{%   endif %}*/
/*{% endif %}*/
}
/*{% endfor %}*/

File: redhawk/codegen/jinja/java/ports/templates/generic.provides.java
//% set classname = portgenerator.className()
package ${package};

import ${component.package}.${component.baseclass.name};

/**
 * @generated
 */
public class ${classname} extends ${portgenerator.poaClass()} {

    /**
     * @generated
     */
    protected ${component.baseclass.name} parent;

    /**
     * @generated
     */
    protected String name;

    /**
     * @generated
     */
    public ${classname}(${component.baseclass.name} parent, String portName)
    {
        this.parent = parent;
        this.name = portName;

        //begin-user-code
        //end-user-code
    }
/*{% for operation in portgenerator.operations() %}*/

    /**
     * @generated
     */
    public ${operation.returns} ${operation.name}(${operation.arglist})${" throws " + operation.throws if operation.throws}
    {
        //begin-user-code
        // TODO you must provide an implementation for this port.
/*{% if operation.returns != 'void' %}*/
    /*{%   if operation.returns == 'org.omg.CORBA.Object' %}*/
        return null;
/*{%   else %}*/
        return ${java.defaultValue(operation.returns)};
/*{%   endif %}*/
/*{% endif %}*/
        //end-user-code
    }
/*{% endfor %}*/
}
